File: scm-java/src/main/scheme/primitives/PrimitiveThreadSpecific.java
package scheme.primitives;
import scheme.*;

public class PrimitiveThreadSpecific extends Primitive {
    @Override
    public String name() { return "thread-specific"; }

    @Override
    public String info() {
        return "Syntax: (thread-specific thread)\n" +
               "Library: (srfi 18)\n" +
               "Description: Returns the thread-specific data of the thread.\n" +
               "Example:\n" +
               "  (thread-specific (current-thread)) => ()";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        SchemeThread t = (SchemeThread) Value.asNativeValue(arguments[0]).value;
        return t.specific;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveThreadSpecific.cs
namespace scheme;

public class PrimitiveThreadSpecific : Primitive
{
    public override string Name() => "thread-specific";

    public override string Info() =>
        "Syntax: (thread-specific thread)\n" +
        "Library: (srfi 18)\n" +
        "Description: Returns the thread-specific data associated with the thread.\n" +
        "Example:\n" +
        "  (thread-specific (current-thread)) => ()";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        SchemeThread t = (SchemeThread) Value.AsNativeValue(arguments[0]).value;
        return t.specific;
    }
}
